### PR TITLE
Yanbo/petlisting

### PR DESCRIPTION
**Pull request by iamyanbo**
locally tested:
- Notifications only gets added to users with seekers with notifications true
- Tested only creator of petlisting is able to update and delete it
- Tested adding pictures, currently stores them to media which has been added to gitignore
- Tested when petlisting doesn't exist throw 400 error
- Tested when non-owner tries to modify a petlisting throws 401 error

Currently the paginated petlisting when a user tries to get all will only return 2 per page for demo purposes
And also updated notifications to use pages with 4 per page

### CODE REVIEW

File: petpal/comments/models.py [hunk 1]
```diff
@@ -27,18 +27,18 @@ def save(self, *args, **kwargs):
         body = ''
         if self.is_review:
             if self.is_author_seeker:
-                subject = 'You have received a review!'
-                body = f'You have received a review from {self.seeker.email}.'
+                subject = f'You have received a review from {self.seeker.email}!'
```
Comment (iamyanbo) [inline petpal/comments/models.py:30]: For ease of access in FE

File: petpal/notifications/models.py [hunk 2]
```diff
@@ -12,6 +12,7 @@ class Notification(models.Model):
     subject = models.TextField()
     body = models.TextField(blank=True)
     recipients = models.ManyToManyField(User, related_name="notifications")
+    recipients_read = models.ManyToManyField(User, related_name="notifications_read", blank=True)
```
Comment (iamyanbo) [inline petpal/notifications/models.py:15]: A2 requires we are able to filter by read and not read, so this is added

File: petpal/notifications/views.py [hunk 3]
```diff
@@ -20,52 +20,81 @@
...
-        data.append({
+        result = {
             'id': notification.pk,
             'subject': notification.subject,
             'body': notification.body,
             'content_type': notification.content_type.model,
             'object_id': notification.object_id,
-        })
+        }
+        if notification.content_type.model == 'petlisting':
```
Comment (iamyanbo) [inline petpal/notifications/views.py:44]: A2 requires a status on the notification

File: petpal/notifications/views.py [hunk 4]
```diff
@@ -15,53 +16,85 @@
...
 @api_view(['GET'])
+@permission_classes([IsAuthenticated])
 def notifications_list_view(request):
-    if not request.user.is_authenticated:
-        return JsonResponse({'data': 'User not logged in'}, status=status.HTTP_401_UNAUTHORIZED)
-    notifications = Notification.objects.filter(recipients=request.user)
+    paginator = PageNumberPagination()
+    paginator.page_size = 4
+    is_read = request.GET.get('is_read', None)
+    print(is_read)
```
Comment (lujorryn) [inline petpal/notifications/views.py]: Please remove this

**Review by lujorryn — APPROVED** [review]
Approving for now, please go to #4 and check off all relative tasks that are confirmed working before merging.
Extra print statement hanging, should remove that before merging as well.

**Pull request merged**